app/assets/javascripts/dashboard/dashboard.js

```diff
@@ -508,6 +566,30 @@ function removePendingAcceptance(minionId) {
   sessionStorage.removeItem(minionId);
 }
 
+function hasPendingRejection(minionId) {
+  return sessionStorage.getItem('reject-'+minionId) === 'true';
+}
+
+function setPendingRejection(minionId) {
+  sessionStorage.setItem('reject-'+minionId, true);
+  $('.pending-nodes-container a[data-minion-id="' + minionId + '"]').parent().text('Rejection in progress');
+}
+
+function removePendingRejection(minionId) {
+  sessionStorage.removeItem('reject-'+minionId);
+}
+
+function setPendingRemoval(minionId) {
```

Comment:
We also forgot to disable the other `Remove` links, so adding 

```js
$('.remove-minion').addClass('disabled');
```

would do the trick.

Reply:
Remember to do the opposite on the `removePendingRemoval` method.